scripts/version.go

```diff
@@ -0,0 +1,193 @@
+// run as 'go run ./scripts/version.go'
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"os/user"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/semver"
+)
+
+func logAndExit(err error) {
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func main() {
+	flag.Usage = func() {
+		fmt.Fprintf(flag.CommandLine.Output(), "Outputs the current version number for this repo\nUsage:\n")
+		flag.PrintDefaults()
+	}
+	goflagsVar := flag.Bool("g", false, "generate Go flags instead of just version")
+	next := flag.Bool("next", false, "list the next version for tagging")
+	releaseTag := flag.Bool("release", false, "tag a new release, and push it")
+	nosign := flag.Bool("nosign", false, "don't sign release tags (use with -release)")
+	flag.Parse()
+
+	ver, err := version()
+	logAndExit(err)
+
+	switch {
+	case *next:
+		nextVer := ver.IncPatch()
+		fmt.Println((&nextVer).String())
+	case *releaseTag:
+		err := release(ver, *nosign)
+		logAndExit(err)
+	case !*goflagsVar:
+		fmt.Printf("%s\n", ver)
+	default:
```

Comment:
When/why do we call this with the `goFlagsVar` flag?

Reply:
here: https://github.com/grafana/smtprelay/pull/45/files/6a21a6fc71409774814c96fe1b9dd898734b9ef7#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R7